### PR TITLE
Fix load failed by display loaded

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes are proposed in this pull request?

Load failed but display loaded.

### Why are the changes needed?

Fix bug.

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including

No

### CODE REVIEW

**Comment by maobaolong** [comment]
@dbw9580 Would you please take a look at this PR?

File: shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java [hunk 1]
```diff
@@ -144,8 +147,11 @@ private void runLoadTask(AlluxioURI filePath, URIStatus status, boolean local)
       }
       Protocol.OpenUfsBlockOptions openUfsBlockOptions =
           new InStreamOptions(status, options, conf).getOpenUfsBlockOptions(blockId);
-      cacheBlock(blockId, dataSource, status, openUfsBlockOptions);
+      if (!cacheBlock(blockId, dataSource, status, openUfsBlockOptions)) {
+        return false;
+      }
```
Comment (dbw9580) [inline shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java:152]: Will this prevent the other blocks from loading, if one of the blocks fails to be cached?

File: shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java [hunk 2]
```diff
@@ -179,9 +185,11 @@ private void cacheBlock(long blockId, WorkerNetAddress dataSource, URIStatus sta
     try (CloseableResource<BlockWorkerClient> blockWorker =
         mFsContext.acquireBlockWorkerClient(dataSource)) {
       blockWorker.get().cache(request);
+      return true;
     } catch (Exception e) {
       System.out.printf("Failed to complete cache request for block %d of file %s: %s", blockId,
           status.getPath(), e);
+      return false;
```
Comment (dbw9580) [inline shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java:192]: I prefer using exceptions to propagate the result of the operation if it is a failure. We can do:

```suggestion
    try (CloseableResource<BlockWorkerClient> blockWorker =
        mFsContext.acquireBlockWorkerClient(dataSource)) {
      blockWorker.get().cache(request);
    } catch (Exception e) {
      throw new RuntimeException(String.format("Failed to complete cache request for block %d of file %s: %s", blockId,
          status.getPath(), e), e);
     }
```

and catch the exception inside the `load` method, which should be the sole method responsible for converting exceptions into error messages and printing them out.

**Review by dbw9580 — CHANGES_REQUESTED** [review]

File: shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java (hunk 1 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java:152]: yeah, do you think this make sense?

File: shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java [hunk 3]
```diff
@@ -180,8 +186,9 @@ private void cacheBlock(long blockId, WorkerNetAddress dataSource, URIStatus sta
         mFsContext.acquireBlockWorkerClient(dataSource)) {
       blockWorker.get().cache(request);
     } catch (Exception e) {
-      System.out.printf("Failed to complete cache request for block %d of file %s: %s", blockId,
-          status.getPath(), e);
+      throw new RuntimeException(String.format("Failed to complete cache request for block %d of file %s: %s", blockId,
+          status.getPath(), e), e);
+     }
```
Comment (maobaolong) [inline shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java:191]:
```suggestion
     }
     return true
```

**Comment by maobaolong** [comment]
@dbw9580 Thanks for your suggestion, PTAL.

File: shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java [hunk 4]
```diff
@@ -180,8 +180,8 @@ private void cacheBlock(long blockId, WorkerNetAddress dataSource, URIStatus sta
         mFsContext.acquireBlockWorkerClient(dataSource)) {
       blockWorker.get().cache(request);
     } catch (Exception e) {
-      System.out.printf("Failed to complete cache request for block %d of file %s: %s", blockId,
-          status.getPath(), e);
+      throw new RuntimeException(String.format("Failed to complete cache request"
```
Comment (rongrong) [inline shell/src/main/java/alluxio/cli/fs/command/LoadCommand.java:183]: Need an extra space between "request" and "for".

**Comment by rongrong** [comment]
Is there people using the load command?

**Comment by maobaolong** [comment]
@rongrong Thanks for your review. 

Is there people using the load command?
------------
Yeah, we use load command besides distributedLoad, as the load command can be more lightweight for small file, meanwhile, there are some performance and stable issue for distributedload.

**Comment by maobaolong** [comment]
@dbw9580 @rongrong Would you like take another look?

**Review by dbw9580 — COMMENTED** [review]
LGTM

**Review by dbw9580 — APPROVED** [review]

**Comment by maobaolong** [comment]
@rongrong Would you like to take another look at this PR? Thanks.

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**